sssd_test_framework/roles/generic.py

```diff
@@ -127,6 +129,35 @@ def fqn(self, name: str) -> str:
         """
         pass
 
+    @abstractmethod
+    def dns(self) -> GenericDNSServer:
+        """
+        Get DNS server object.
+
+            Get methods use dig and is parsed by jc. The data from jc contains several nested dict,
+            but two are returned as a tuple, ``answer, authority``.
+
+        .. code-block:: python
+            :caption: Example usage
+
+            # Create forward zone, add and get record and assert ip
+            forward_zone = provider.dns().zone("smb.test").create()
+            forward_zone.add_record("client", "172.16.200.5")
```

Comment:
A `--create-reverse` argument would be nice here.

Reply:
Sure, I don't insist however just a nice to have IMO.

Reply:
I found a pretty intuitive way to do it. ```add_record(name: str, data: str | int): ``` , int = ptr add, str = a/aaaa add